### PR TITLE
Global navigation autocomplete

### PR DESCRIPTION
**Pull request by henryyang42**
Add global navigation autocomplete feature.
By pressing enter or clicking them can go to the desired page.
Now, it support for
- Problem
- Contest
- Group
- User

### CODE REVIEW

File: index/views.py [hunk 1]
```diff
@@ -34,9 +35,20 @@
...
+from problem.models import Problem
+from contest.models import Contest
+from group.models import Group
 
 # Create your views here.
 logger = get_logger()
+
+
+
+def gg(request):
```
Comment (nickelcheng) [inline index/views.py]: Please use another function name QQ

File: index/urls.py [hunk 2]
```diff
@@ -25,4 +25,6 @@
     url(r'^$', views.index, name='index'),
     url(r'^index/(?P<alert_info>\w+)/$', views.index, name='alert'),
     url(r'^get_time/$', views.get_time),
+    url(r'^search/$', views.navigation_autocomplete, name='search'),
+    url(r'^gg/$', views.gg),
```
Comment (nickelcheng) [inline index/urls.py]: zz

File: index/templates/index/navigation_autocomplete.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{% if problems %}
```
Comment (bruce3557) [inline index/templates/index/navigation_autocomplete.html]: Fine with this.
If having performance issue, I think we can just add license in the README.md.
Just decreasing some security

**Comment by bruce3557** [comment]
+1 besides the issue

File: index/templates/index/navigation_autocomplete.html (hunk 3 above)
Comment (nickelcheng) [inline index/templates/index/navigation_autocomplete.html]: @henryyang42 can you open an issue for this?

**Comment by nickelcheng** [comment]
going to merge

**Pull request merged**